lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -423,8 +423,11 @@ def redeemable_shares(entity)
 
         def emergency_issuable_bundles(entity)
           return [] if @round.issued_shares[entity]
+          return [] if entity.all_abilities.find { |a| a.type == :train_discount } && entity.cash >= @depot.min_depot_price * 0.9
```

Comment:
You're splitting the game logic up: for most cases you're checking whether the corporation can afford a train in `Step::BuyTrain.can_afford_neeed_train?`. But if the corporation has the discount ability you test against the reduced price here.

It would make the logic clearer if you instead modified `can_afford_needed_train?` to test against the discounted price.

Reply:
resolved